### PR TITLE
Added some slack to required gas of relayed call

### PR DESCRIPTION
**Pull request by shahafn**
Since the on-chain calculation is too tight. It seeems as if the size of the encodedFunction might be responsible for this, So I added a calculation per the yellowpaper for its gas cost and added it to the relayed tx's gas limit.

Also removed duplicate line in test removed unused params

### CODE REVIEW

File: server/src/librelay/relay_server.go [hunk 1]
```diff
@@ -355,9 +347,6 @@ func (relay *RelayServer) RegistrationDate() (when int64, err error) {
 	if (iter.Event == nil && !iter.Next()) ||
 		(bytes.Compare(iter.Event.Relay.Bytes(), relay.Address().Bytes()) != 0) ||
 		(iter.Event.TransactionFee.Cmp(relay.Fee) != 0) ||
-		(iter.Event.Stake.Cmp(relay.StakeAmount) < 0) ||
```
Comment (drortirosh) [inline server/src/librelay/relay_server.go]: I'm trying to undertstand what did this line meant...

File: server/src/librelay/relay_server.go [hunk 2]
```diff
@@ -486,6 +475,8 @@ func (relay *RelayServer) CreateRelayTransaction(request RelayTransactionRequest
 		log.Println(err)
 		return
 	}
+	// Since the on chain calculation is too tight and sometime insufficient, we need to give some slack
```
Comment (drortirosh) [inline server/src/librelay/relay_server.go]: actually, the comment is not relevant, its not just adding slack, is properly calculating gas used.

File: server/src/librelay/relay_server_test.go [hunk 3]
```diff
@@ -81,41 +82,41 @@ type TestServer struct {
 	*RelayServer
 }
 
-func (relay *TestServer) Stake(ownerKey *ecdsa.PrivateKey) (err error) {
-	tx, err := relay.sendStakeTransaction(ownerKey)
+func (relay *TestServer) Stake(ownerKey *ecdsa.PrivateKey, stakeAmount *big.Int, unstakeDelay *big.Int) (err error) {
```
Comment (drortirosh) [inline server/src/librelay/relay_server_test.go]: a. not sure what you gained by removing the fields, if you now need to move them manually around to methods.
b. clearly its not a tiny cleanup, and doesn't fit into the gas-price fix.

File: server/src/librelay/relay_server_test.go [hunk 4]
```diff
@@ -128,6 +129,8 @@ var client *TestClient
 var relayKey1 *ecdsa.PrivateKey
 var gaslessKey2 *ecdsa.PrivateKey
 var ownerKey3 *ecdsa.PrivateKey
+var unstakeDelay = big.NewInt(1 * 60 * 60 * 24 * 7) // 1 week in seconds
```
Comment (drortirosh) [inline server/src/librelay/relay_server_test.go]: it bothers me to see these constants... did we ever test that a relay works with DIFFERENT values for stake delay/amount ?

**Comment by drortirosh** [comment]
also, you're changing the semantics a bit. change the version to `0.4.1`

File: server/src/librelay/relay_server_test.go (hunk 3 above)
Comment (shahafn) [reply server/src/librelay/relay_server_test.go]: These methods are just in the tests, they aren't called anywhere else so no reason to keep fields just for methods defined and used only in a test file.

File: server/src/librelay/relay_server_test.go (hunk 4 above)
Comment (shahafn) [reply server/src/librelay/relay_server_test.go]: It looks like we do in `RelayHub.test.js`

File: server/src/librelay/relay_server.go (hunk 1 above)
Comment (shahafn) [reply server/src/librelay/relay_server.go]: I was validating the values of the staked event but it doesn't really add much and can be removed.

File: server/src/librelay/relay_server.go (hunk 2 above)
Comment (shahafn) [reply server/src/librelay/relay_server.go]: Fixed

**Comment by shahafn** [comment]
> Approved too early. Need to update the version

Updated

**Comment by drortirosh** [comment]
@nventuro : please upgrade your relays to this version (0.4.1).

**Pull request merged**